app/controllers/admin/customers_controller.rb

```diff
@@ -66,7 +66,12 @@ def collection
       return Customer.where("1=0") unless json_request? && params[:enterprise_id].present?
 
       Customer.of(managed_enterprise_id).
-        includes(:bill_address, :ship_address, user: :credit_cards)
+        includes(:bill_address, :ship_address, user: :credit_cards).
+        joins(:orders).
+        merge(Spree::Order.complete.not_state(:canceled)).
```

Comment:
So can we just change this line from `.not_state(:cancelled)` to `.where(state: ['complete', 'canceled'])` and get the same result as the pending PR?

Reply:
Ah no, I see. We need to sum all payments on both cancelled and complete orders and sum all totals on complete orders. I'm sure we can manage this in a single query :+1: